sympy/functions/elementary/piecewise.py

```diff
@@ -305,13 +306,115 @@ def doit(self, **hints):
             newargs.append((e, c))
         return self.func(*newargs)
 
+    def is_meromorphic(self, x, a):
+        """
+        This method tests whether a Piecewise expression is meromorphic
+        as a function of the given symbol ``x`` at the point ``a``.
+
+        This method is intended as a quick test that will return
+        None if no decision can be made without simplification or
+        more detailed analysis.
+
+        Examples
+        ========
+        >>> from sympy import csc, log, Piecewise
+        >>> from sympy.abc import x
+        >>> f = Piecewise((csc(1/x), x < 1), (2*x**2 + 1, x < 5), (log(x), True))
+        >>> f.is_meromorphic(x, 0)
+        False
+        >>> f.is_meromorphic(x, 2)
+        True
+        >>> f.is_meromorphic(x, 10)
+        True
+
+        """
+        if not x.is_symbol:
+            raise TypeError("{} should be of symbol type".format(x))
+        a = sympify(a)
+        upper_bound_list, lower_bound_list = [], []
+        for l, u, e, i in self._intervals(x)[1]:
+            upper_bound_list.append(u)
+            lower_bound_list.append(l)
+
+        if a not in lower_bound_list and a not in upper_bound_list:
+            for bound in self.as_expr_set_pairs():
+                if a in bound[1]:
+                    return bound[0]._eval_is_meromorphic(x, a)
+        else:
+            return None
+
     def _eval_simplify(self, **kwargs):
         return piecewise_simplify(self, **kwargs)
 
-    def _eval_as_leading_term(self, x, logx=None, cdir=0):
-        for e, c in self.args:
-            if c == True or c.subs(x, 0) == True:
-                return e.as_leading_term(x)
+    def _eval_as_leading_term(self, x, logx=None, cdir=1):
+        from sympy.sets.sets import Union
+        if cdir == S.Zero:
+            l = self._eval_as_leading_term(x, logx=None, cdir=-1)
+            r = self._eval_as_leading_term(x, logx=None, cdir=1)
+            if l != r:
+                raise ValueError
+            return r
+
+        upper_bound_list, lower_bound_list = [], []
+        for a, b, e, i in self._intervals(x)[1]:
```

Comment:
`_intervals` may fail. That should be checked.

Reply:
Will try to address this too , I see `_intervals` can return `False` too , will return what we already have on master in that case.

Reply:
@jksuom is this the how we should be approaching this ..... cause everytime we encounter `False` for `self._intervals(x)[0]` , a message is returned . As the docstring says for _intervals , we have 
```
If there are any relationals not involving sym, or any
relational cannot be solved for sym, the bool will be False
a message be given as the second return value.
```
The message raised goes like this `'Unable to solve relational %s for %s.' % (r, sym)`
So should the following diff be used ! 
```
--- a/sympy/functions/elementary/piecewise.py
+++ b/sympy/functions/elementary/piecewise.py
@@ -357,6 +357,8 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
             return r

         zero_in_upper_bound, zero_in_lower_bound = None, None
+        if not self._intervals(x)[0]:
+            raise NotImplementedError("%s" % (self._intervals(x)[1]))
         for a, b, e, i in self._intervals(x)[1]:
             if b is S.Zero:
                 zero_in_upper_bound = True
```

Reply:
Also as you had pointed out that the low level fix for the bi-directional case wasn't the best way to approach it . I had also replied to that some days back https://github.com/sympy/sympy/pull/22555#issuecomment-1047394766 and still I am not completely sure how can one go about this . So I wanted your guidance on this too , like how you could you maybe approach this ? If I get some hint of a workaround or so I'll surely try that out !

Reply:
Another doubt I had was why haven't we considered having a separate number for bi-directional cases, like was it ever thought that we could have a sperate number or even `cdir = 0` reserved only for bi-directional cases . It might help us at multiple places !